src/app/api/activity/activityLogger.service.ts

```diff
@@ -0,0 +1,124 @@
+import { ActivityType, AssigneeType, Task } from '@prisma/client'
+import { BaseService } from '@/app/api/core/services/base.service'
+import { CopilotAPI } from '@/utils/CopilotAPI'
+import User from '../core/models/User.model'
+import { ClientResponse, CompanyResponse, InternalUsers, MeResponse } from '@/types/common'
+import { UpdateTaskRequest } from '@/types/dto/tasks.dto'
+import WorkflowStatesService from '../workflow-states/workflowStates.service'
+
+export class ActivityLogger extends BaseService {
+  public taskId: string
+
+  constructor({ taskId, user }: { taskId: string; user: User }) {
+    super(user)
+    this.taskId = taskId
+  }
+
+  async createTaskLog() {
+    const userInfo = await this.getUserInfo(this.user.token)
+
+    await this.db.activityLog.create({
+      data: {
+        taskId: this.taskId,
+        workspaceId: this.user.workspaceId,
+        activityType: ActivityType.CREATE_TASK,
+        details: {
+          initiator: userInfo?.givenName || '' + ' ' + userInfo?.familyName || '',
+          initiatorId: this.user.internalUserId as string,
```

Comment:
validate with zod instead of typecast

Reply:
Updated in dbbb36dc566d6f2ce2ade14c66328c03c8b5ece0 commit